pymc3/distributions/distribution.py

```diff
@@ -148,17 +148,16 @@ def default(self):
     def get_test_val(self, val, defaults):
         if val is None:
             for v in defaults:
-                if hasattr(self, v) and np.all(np.isfinite(self.getattr_value(v))):
-                    return self.getattr_value(v)
-        else:
-            return self.getattr_value(val)
-
-        if val is None:
+                attr_val = self.getattr_value(v) if hasattr(self, v) else None
```

Comment:
I think that with the removed if block we can turn this one into two simple nested ifs
```python
if hasattr():
  attr_val = ...
  if np.all(np.isfinite(attr_val)):
    return attr_val